### ISSUE TITLE
Flutter 3 fix

### ISSUE THREAD

**Issue by fennelhans** [Tkko/Flutter_Pinput#89]
In Flutter 3, `WidgetsBinding.instance` no longer requires a null safety check.

```dart
: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
../…/src/pinput_state.dart:81
- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../../../bin/flutter/packages/flutter/lib/src/widgets/binding.dart').
package:flutter/…/widgets/binding.dart:1
    WidgetsBinding.instance!.addObserver(this);
                   ^
: Warning: Operand of null-aware operation '!' has type 'WidgetsBinding' which excludes null.
../…/src/pinput_state.dart:202
- 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../../../bin/flutter/packages/flutter/lib/src/widgets/binding.dart').
package:flutter/…/widgets/binding.dart:1
    WidgetsBinding.instance!.removeObserver(this);
```

**Comment by Tkko** [comment]
Hi @fennelhans, I'm merging the PR RN.

**Comment by fennelhans** [comment]
Thanks!

**Comment by Tkko** [comment]
@fennelhans, this won't work on old Flutter version.

```
 Error: Method 'addObserver' cannot be called on 'WidgetsBinding?' because it is potentially null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../Tools/flutter/packages/flutter/lib/src/widgets/binding.dart').
Try calling using ?. instead.
    WidgetsBinding.instance.addObserver(this);
```

**Comment by fennelhans** [comment]
Yeah, you'll probably have to bump the supported version of dart

**Comment by AlexV525** [comment]
You can set the new version as `2.3.0`, then set the `sdk` constraint to `>=2.17.0 <3.0.0`, and `flutter` to `>=3.0.0`. This won't break anyone since they won't able to fetch the new version until they upgrade the Flutter.

**Comment by Tkko** [comment]
Just found the better way

https://github.com/Tkko/Flutter_Pinput/blob/d535b90eedce418ed43eafe38b2965b1e1831ac3/lib/src/pinput_state.dart#L83

**Comment by Tkko** [comment]
Fixed, version [2.2.8](https://pub.dev/packages/pinput)
